src/pkg/api/template_test.go

```diff
@@ -148,7 +149,149 @@ func TestTemplateHandler_GetByID(t *testing.T) {
+								<language>
+									<terminology_id>
+										<value>ISO_639-1</value>
+									</terminology_id>
+									<code_string>en</code_string>
+								</language>
+								<description>
+									<original_author id="Original Author">Not Specified</original_author>
+									<lifecycle_state>Initial</lifecycle_state>
+									<other_details id="MetaDataSet:Sample Set ">Template metadata sample set</other_details>
+									<details>
+										<language>
+											<terminology_id>
+												<value>ISO_639-1</value>
+											</terminology_id>
+											<code_string>en</code_string>
+										</language>
+										<purpose>Not Specified</purpose>
+									</details>
+								</description>
+								<uid>
+									<value>b4d7f203-b329-4e89-a58a-c605b19e94de</value>
+								</uid>
+								<template_id>
+									<value>` + template.TemplateID + `</value>
+								</template_id>
+								<concept>` + template.Concept + `</concept>
+								<definition archetype_id="` + template.ArchetypeID + `"
+									concept_name="Encounter" name="vital_signs2">
+									<rm_type_name>COMPOSITION</rm_type_name>
+									<occurrences>
+										<lower_included>true</lower_included>
+										<upper_included>true</upper_included>
+										<lower_unbounded>false</lower_unbounded>
+										<upper_unbounded>false</upper_unbounded>
+										<lower>1</lower>
+										<upper>1</upper>
+									</occurrences>
+									<node_id>at0000</node_id>
+								</definition>
+							</template>
+		`)
+
+	tests := []struct {
+		name       string
+		body       string
+		adlVer     string
+		prepare    func(gaSvc *mocks.MockTemplateService)
+		wantStatus int
+	}{
+		{
+			"1. empty result because body is empty",
+			"",
+			model.VerADL1_4,
+			func(gaSvc *mocks.MockTemplateService) {},
+			http.StatusBadRequest,
+		},
+		{
+			"2. empty result because body is not valid",
+			"<xml>...ups",
+			model.VerADL1_4,
+			func(gaSvc *mocks.MockTemplateService) {
+				gaSvc.EXPECT().Parser(model.VerADL1_4).Return(adl14.NewParser(), nil)
+			},
+			http.StatusBadRequest,
+		},
+		{
+			"3. success",
+			string(template.Body),
+			model.VerADL1_4,
+			func(gaSvc *mocks.MockTemplateService) {
+				gaSvc.EXPECT().Parser(model.VerADL1_4).Return(adl14.NewParser(), nil)
+				gaSvc.EXPECT().Store(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil)
+			},
+			http.StatusCreated,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			tSvc := mocks.NewMockTemplateService(ctrl)
+			tt.prepare(tSvc)
+
+			// Mock for auth user service
+			userSvc := mocks.NewMockUserService(ctrl)
+			userSvc.EXPECT().VerifyAccess(gomock.Any(), gomock.Any()).Return(nil)
+
+			api := API{
+				Template: NewTemplateHandler(tSvc, ""),
+				User:     NewUserHandler(userSvc),
+			}
+
+			router := api.setupRouter(api.buildDefinitionAPI())
```

Comment:
We can take API building outside the loop. 
Then it won't build every time and the log will be cleaner

Reply:
It does not take a long time.  And if you set up mocks outside for loop, it would affect other test cases.